public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,210 @@
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { Alert, Button, LoadingPlaceholder, useStyles2 } from '@grafana/ui';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import ResourcePickerData from '../../resourcePicker/resourcePickerData';
+import messageFromError from '../../utils/messageFromError';
+import { Space } from '../Space';
+import NestedResourceTable from './NestedResourceTable';
+import { ResourceRow, ResourceRowGroup, ResourceRowType } from './types';
+import { addResources, findRow, parseResourceURI } from './utils';
+
+interface ResourcePickerProps {
+  resourcePickerData: ResourcePickerData;
+  resourceURI: string | undefined;
+  templateVariables: string[];
+
+  onApply: (resourceURI: string | undefined) => void;
+  onCancel: () => void;
+}
+
+const ResourcePicker = ({
+  resourcePickerData,
+  resourceURI,
+  templateVariables,
+  onApply,
+  onCancel,
+}: ResourcePickerProps) => {
+  const styles = useStyles2(getStyles);
+
+  type LoadingStatus = 'NotStarted' | 'Started' | 'Done';
+  const [loadingStatus, setLoadingStatus] = useState<LoadingStatus>('NotStarted');
+  const [azureRows, setAzureRows] = useState<ResourceRowGroup>([]);
+  const [internalSelected, setInternalSelected] = useState<string | undefined>(resourceURI);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+
+  // Sync the resourceURI prop to internal state
+  useEffect(() => {
+    setInternalSelected(resourceURI);
+  }, [resourceURI]);
+
+  // Request initial data on first mount
+  useEffect(() => {
+    if (loadingStatus === 'NotStarted') {
+      const loadInitialData = async () => {
+        try {
+          setLoadingStatus('Started');
+          let resources = await resourcePickerData.getSubscriptions();
+          if (!internalSelected) {
+            setAzureRows(resources);
+            setLoadingStatus('Done');
+            return;
+          }
+
+          const parsedURI = parseResourceURI(internalSelected ?? '');
+          const resourceGroupURI = `/subscriptions/${parsedURI?.subscriptionID}/resourceGroups/${parsedURI?.resourceGroup}`;
```

Comment:
thinking out loud but what if `internalSelected` is a subscription?

Reply:
I see that it won't enter in the other `if`s. Nit but for readability I would write:

```
const parsedURI = parseResourceURI(internalSelected ?? '');
if (parsedURI?.resourceGroup) {
             const resourceGroupURI = `/subscriptions/${parsedURI?.subscriptionID}/resourceGroups/${parsedURI?.resourceGroup}`;
             ....
```